.github/workflows/first-interaction.yaml

```diff
@@ -0,0 +1,85 @@
+# Summary: add a welcoming comment to first-time contributors' issues & PRs.
+# This is written in a generic way so that we can use the same workflow
+# in all our quantumlib repos.
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+name: Welcome first interactions
+run-name: Welcome ${{github.actor}}'s first interaction
+
+on:
+  issues:
+    types: [opened]
+  pull_request:
+    types: [opened]
+
+# Declare default permissions as read-only.
+permissions: read-all
+
+jobs:
+  welcome:
+    name: Check for first interaction
+    runs-on: ubuntu-24.04
+    timeout-minutes: 10
+    permissions:
+      issues: write
+      pull-requests: write
+    env:
+      repo: ${{github.server_url}}/${{github.repository}}
+      files: ${{github.server_url}}/${{github.repository}}/blob/${{github.ref_name}}
+    steps:
+      - name: Check out a copy of the git repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+
+      - name: Add a welcome comment if appropriate
+        uses: actions/first-interaction@2ec0f0fd78838633cd1c1342e4536d49ef72be54 # v1.3.0
+        with:
+          repo-token: ${{secrets.GITHUB_TOKEN}}
+          issue-message: |
```

Comment:
does this trigger on issues as well? or just PRs?

Reply:
Hi – both issues and PRs. Or more specifically, when either is opened. The conditions are in the `on:` block near the top:

```yaml
on:
  issues:
    types: [opened]
  pull_request:
    types: [opened]
```